src/test/java/org/otherband/email/smtp/SMTPEmailServiceTest.java

```diff
@@ -0,0 +1,30 @@
+package org.otherband.email.smtp;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.otherband.email.VerificationLinkEmailRequest;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+
+import static org.mockito.Mockito.*;
+
+class SMTPEmailServiceTest  {
+    private SMTPEmailService service;
+    private String mailUserName;
+    private JavaMailSender mailSender;
+
+    @BeforeEach
+    void setup() {
+        mailUserName = "test";
+        mailSender = mock(JavaMailSender.class);
+        service = new SMTPEmailService(mailUserName, mailSender);
+    }
+
+    @Test
+    void testSendVerificationEmail() {
+        VerificationLinkEmailRequest verificationLinkEmailRequest = new VerificationLinkEmailRequest();
+        service.sendLetterVerificationLink(verificationLinkEmailRequest);
+        verify(mailSender).send(Mockito.any(SimpleMailMessage.class));
+    }
```

Comment:
Can we add content to the VerificationLinkEmailRequest and verify the fields of the created SimpleMailMessage

Reply:
How do I access the created SimpleMailMessage in this context when it's local to the sendLetterVerificationLink method? should we have the method return it?

Reply:
You can use Mockito's ArgumentCaptor to capture the SimpleMailMessage

Reply:
resolved